lib/src/models/retry_event_model.dart

```diff
@@ -0,0 +1,26 @@
+import 'dart:async';
+
+/// Retries the `retryFunction` after a set `timeInterval` until `dispose` is called
+class RetryEventModel {
+  /// Time interval in milliseconds
+  Duration timeInterval;
+  void Function(Timer timer) retryFunction;
```

Comment:
this variable seems unnecessary?

Reply:
ye but timer.periodic callback is of time Function(timer), hence the useless vairable